devicetypes/smartthings/viconics-schneider-room-controller.src/viconics-schneider-room-controller.groovy

```diff
@@ -0,0 +1,549 @@
+	}
+}
+
+def auto() {
+	getThermostatModeCommands(THERMOSTAT_MODE_AUTO)
+}
+
+def cool() {
+	getThermostatModeCommands(THERMOSTAT_MODE_COOL)
+}
+
+def heat() {
+	getThermostatModeCommands(THERMOSTAT_MODE_HEAT)
+}
+
+def off() {
+	getThermostatModeCommands(THERMOSTAT_MODE_OFF)
+}
+
+def getThermostatModeCommands(mode) {
+	if (mode) {
+		delayBetween([
+				zigbee.writeAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_MODE, DataType.ENUM8, mode),
+				zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_MODE),
+				zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_OPERATING_STATE)
+		], 500)
+	}
+}
+
+def ping() {
+	log.debug "ping"
+	zigbee.readAttribute(THERMOSTAT_CLUSTER, LOCAL_TEMPERATURE)
+}
+
+def refresh() {
+	log.debug "refresh"
+	getRefreshCommands()
+}
+
+def getRefreshCommands() {
+	def refreshCommands = []
+
+	refreshCommands += zigbee.readAttribute(RELATIVE_HUMIDITY_CLUSTER, RELATIVE_HUMIDITY_MEASURED_VALUE)
+	refreshCommands += zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_HUMIDITY)
+
+	refreshCommands += zigbee.readAttribute(THERMOSTAT_CLUSTER, LOCAL_TEMPERATURE)
+	refreshCommands += zigbee.readAttribute(THERMOSTAT_CLUSTER, COOLING_SETPOINT)
+	refreshCommands += zigbee.readAttribute(THERMOSTAT_CLUSTER, HEATING_SETPOINT)
+
+	refreshCommands += zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_FAN_SPEED)
+	refreshCommands += zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_FAN_MODE)
+	refreshCommands += zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_MODE) // formerly THERMOSTAT MODE: 0x001C
+	refreshCommands += zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_OPERATING_STATE)
+
+	refreshCommands += zigbee.readAttribute(THERMOSTAT_CLUSTER, OCCUPANCY)
+	refreshCommands += zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_OCCUPANCY)
+	refreshCommands += zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_EFFECTIVE_OCCUPANCY)
+	refreshCommands += zigbee.readAttribute(THERMOSTAT_UI_CONFIGURATION_CLUSTER, TEMPERATURE_DISPLAY_MODE)
+
+	refreshCommands += refreshChild()
+
+	refreshCommands
+}
+
+def refreshChild() {
+	log.debug "refresh child device"
+	def refreshCommands = []
+	refreshCommands += zigbee.readAttribute(THERMOSTAT_CLUSTER, COOLING_SETPOINT_UNOCCUPIED)
+	refreshCommands += zigbee.readAttribute(THERMOSTAT_CLUSTER, HEATING_SETPOINT_UNOCCUPIED)
+
+	refreshCommands
+}
+
+def configure() {
+	log.debug "Configuration"
+
+	configureChild()
+
+	def configurationCommands = []
+
+	// set initial values
+	configurationCommands += setSetpoint(initialCoolingSetpoint, COOLING_SETPOINT)
+	configurationCommands += setSetpoint(initialHeatingSetpoint, HEATING_SETPOINT)
+	configurationCommands += setSetpoint(initialCoolingSetpoint, COOLING_SETPOINT_UNOCCUPIED)
+	configurationCommands += setSetpoint(initialHeatingSetpoint, HEATING_SETPOINT_UNOCCUPIED)
+
+	configurationCommands += zigbee.configureReporting(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_MODE, DataType.ENUM8, 1, 60, 1) //formerly THERMOSTAT MODE: 0x001C
```

Comment:
I just realized that you're setting the max reporting interval for _all_ of these attributes at a minute. That's way too low. That requires them to report every minute. If you're min reporting interval is 1 second and the delta is 1 they'll report every change anyway, so please raise the max reporting interval to at least an hour. 

If the device is sending us like 20 updates every minute it will be an incredibly expensive device to run.

Reply:
@greens Good point (max reporting was set to short period for testing purposes).